From 83dfbcb6ebc8fed5f43c92b763276ec4e4355c0c Mon Sep 17 00:00:00 2001
From: Ariel Elior <ariele@broadcom.com>
Date: Wed, 9 Jan 2013 07:04:35 +0000
Subject: [PATCH 578/858] bnx2x: align define usage to satisfy static checkers

Static checkers complained that the E1H_FUNC_MAX define is used
incorrectly in bnx2x_pretend_func(). The complaint was justified,
although its not a real bug, as the first part of the conditional
protects us in this case (a real bug would happen if a VF tried to
use the pretend func, but there are no VFs in E1H chips).

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Ariel Elior <ariele@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 238268503f306cdc435eff2181e38c08ce412ed5)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
index 21b6c29..f808ed0 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -13404,7 +13404,7 @@ int bnx2x_pretend_func(struct bnx2x *bp, u16 pretend_func_val)
 {
  u32 pretend_reg;
 
- if (CHIP_IS_E1H(bp) && pretend_func_val > E1H_FUNC_MAX)
+ if (CHIP_IS_E1H(bp) && pretend_func_val >= E1H_FUNC_MAX)
   return -1;
 
  /* get my own pretend register */
-- 
1.8.3.1